log_test.go

```diff
@@ -0,0 +1,71 @@
+package marker
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/fatih/color"
+	"github.com/stretchr/testify/assert"
+)
+
+type MockLogOut struct {
+	actualLog string
+}
+
+func (m *MockLogOut) Write(p []byte) (n int, err error) {
+	m.actualLog = string(p)
+	return len(p), nil
+}
+
+func Test_New(t *testing.T) {
+	stdoutMarker := NewStdoutMarker()
+	assert.Equal(t, stdoutMarker.out, os.Stdout)
+
+	mockLogOut := &MockLogOut{}
+	stdoutMarker = NewStdoutMarker(SetOutput(mockLogOut))
+	assert.Equal(t, stdoutMarker.out, mockLogOut)
+}
+
+func Test_AddRule(t *testing.T) {
+	stdoutMarker := NewStdoutMarker()
+	markRule := MarkRule{MatchAll("test"), color.New(color.FgRed)}
+	stdoutMarker.AddRule(markRule)
+	assert.Len(t, stdoutMarker.rules, 1)
```

Comment:
is it overkill to test the object in the rules is what we expect it to be?

Reply:
assertion fails when i do that.
```
assert.Equal(t, stdoutMarker.rules[0], markRule)
```
Output:
```
expected: marker.MarkRule{Matcher:(marker.MatcherFunc)(0x6a1780), Color:(*color.Color)(0xc0000af200)}
actual  : marker.MarkRule{Matcher:(marker.MatcherFunc)(0x6a1780), Color:(*color.Color)(0xc0000af200)}
```
Most probably because of, in `testify/assert` there is a statement for `assert.Equal`: 
> Function equality cannot be determined and will always fail. 

For ensuring:
```
assert.Equal(t, markRule.Matcher, stdoutMarker.rules[0].Matcher)
```
Output:
```
Error:      	Invalid operation: (marker.MatcherFunc)(0x6a1710) == (marker.MatcherFunc)(0x6a1710) (cannot take func type as argument)
```

Reply:
Just found a way to assert function equality with a little spice of `reflect`. :bowtie:

Reply:
You can not compare function, nor you can take their address - it is prohibited in standard. You can wrap it in interface, I think.